File: 1287-element-appearing-more-than-25-in-sorted-array/1287-element-appearing-more-than-25-in-sorted-array.py
class Solution:
    def findSpecialInteger(self, arr: List[int]) -> int:
        per = len(arr)//4
        d={}
        for i in arr:
            if d.get(i) == None:
                d[i] = 1
            else:
                d[i]+=1
            if(d[i] > per):
                return i
        return 0

File: 1287-element-appearing-more-than-25-in-sorted-array/1287-element-appearing-more-than-25-in-sorted-array.java
class Solution {
    public int findSpecialInteger(int[] arr) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int per = arr.length /4;
        for(int ele:arr){
            if (map.get(ele) == null){
                map.put(ele,1);
            }else{
                int val = map.get(ele) +1;
                map.put(ele,val);
            }
            if(map.get(ele)>per){
                return ele;
            }
        }
        return 0;
    }
}